src/test/java/com/artipie/FilesRepoITCase.java

```diff
@@ -0,0 +1,163 @@
+import com.artipie.asto.test.TestResource;
+import com.jcabi.log.Logger;
+import java.nio.file.Path;
+import java.util.regex.Pattern;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.core.StringContains;
+import org.hamcrest.text.MatchesPattern;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledOnOs;
+import org.junit.jupiter.api.condition.OS;
+import org.junit.jupiter.api.io.TempDir;
+import org.testcontainers.Testcontainers;
+import org.testcontainers.containers.GenericContainer;
+
+/**
+ * Integration tests for Files repository.
+ * @since 0.11
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+@EnabledOnOs({OS.LINUX, OS.MAC})
+final class FilesRepoITCase {
+
+    /**
+     * Temporary directory for all tests.
+     * @checkstyle VisibilityModifierCheck (3 lines)
+     */
+    @TempDir
+    Path tmp;
+
+    /**
+     * Tested Artipie server.
+     */
+    private ArtipieServer server;
+
+    /**
+     * Container.
+     */
+    private GenericContainer<?> cntn;
+
+    /**
+     * Storage.
+     */
+    private Storage storage;
+
+    /**
+     * Artipie server port.
+     */
+    private int port;
+
+    @BeforeEach
+    void init() throws Exception {
+        this.storage = new FileStorage(this.tmp);
+        this.server = new ArtipieServer(this.tmp, "my-file", this.configs());
+        this.port = this.server.start();
+        this.server.start();
+        Testcontainers.exposeHostPorts(this.port);
+        this.cntn = new GenericContainer<>("centos:centos8")
+            .withCommand("tail", "-f", "/dev/null")
+            .withWorkingDirectory("/home/")
+            .withFileSystemBind(this.tmp.toString(), "/home");
+        this.cntn.start();
+        this.exec("yum", "-y", "install", "curl");
+        this.exec("yum", "-y", "install", "maven");
+    }
+
+    @Test
+    void downloadsArtifact() throws Exception {
+        final String url = "http://host.testcontainers.internal:%d/my-file/helloworld-src/pom.xml";
+        this.addFilesToStorage(
+            "helloworld-src", new Key.From("my-file", "helloworld-src")
+        );
+        MatcherAssert.assertThat(
+            "curl PUT doesn't work properly",
+            this.exec("curl", "-i", "-X", "PUT", String.format(url, this.port)),
+            new StringContains("HTTP/1.1 201 Created")
+        );
+        MatcherAssert.assertThat(
+            "curl GET doesn't work properly",
+            this.exec("curl", "-i", "-X", "GET", String.format(url, this.port)),
+            new MatchesPattern(
+                Pattern.compile(
+                    // @checkstyle LineLengthCheck (1 line)
+                    "HTTP\\/1.1 200 OK[\\r\\n]{0,2}Content-Type: application\\/octet-stream[\\r\\n \\S]*"
+                )
+            )
+        );
+        this.exec("mvn", "clean");
```

Comment:
@genryxy running `mvn clean` command here is not needed, please remove it

Reply:
@olegmoz in some case I get the error as https://github.com/artipie/artipie/pull/537/checks?check_run_id=1081335703#step:5:1028 In the previous case running `mvn clean` command helped to fix. I can try to push without maven

Reply:
@genryxy that was in Maven test, this test is absolutely unrelated

Reply:
@olegmoz understood, thanks